airflow/sensors/external_task_sensor.py

```diff
@@ -40,6 +40,10 @@ class ExternalTaskSensor(BaseSensorOperator):
     :type external_task_id: str
     :param allowed_states: list of allowed states, default is ``['success']``
     :type allowed_states: list
+    :param unallowed_states: list of unallowed states, default is ``None``. When
```

Comment:
```suggestion
    :param failure_states: list of states to treat as a failure, default is ``None``. When
```

Reply:
I think we do need to specify both separately because the goal is to change the sensor from an operator that does one of two things (either succeed or continue to poke) to an operator that does one of three things (succeed, continue to poke, or fail outright).

If we specify `unallowed_states` as `NOT allowed_states`, then the operator still only performs one of two actions (succeed or fail outright, with the fail outright action replacing the original continue to poke).

as for the wording, I do think ashb's wording will make it more clear and will update

Reply:
Thanks. I got it now. I like Ash's suggestion.

Reply:
@michaelmdeng can you apply Ash's suggestion?